store/src/java/com/zimbra/cs/service/mail/CopyToFiles.java

```diff
@@ -55,120 +54,183 @@ public CopyToFiles(
+                    .onFailure(ex -> mLog.debug(ex.getMessage()))
+                    .mapFailure(
+                        Case(
+                            $(ex -> !(ex instanceof ServiceException)),
+                            ex -> ServiceException.FAILURE("internal error.", ex)))
+                    .get())
+            .orElseThrow(() -> ServiceException.FAILURE("got null response from Files server."))
+            .getNodeId();
 
     CopyToFilesResponse copyToFilesResponse = new CopyToFilesResponse();
     copyToFilesResponse.setNodeId(nodeId);
     return zsc.jaxbToElement(copyToFilesResponse);
   }
 
   /**
-   * Transforms the SOAP request in object representation
+   * Transforms the SOAP request in {@link CopyToFilesRequest} instance
    *
    * @param request soap {@link Element} as received from client
    * @return try of {@link CopyToFilesRequest}
    */
   private Try<CopyToFilesRequest> getRequestObject(Element request) {
     return Try.<CopyToFilesRequest>of(() -> JaxbUtil.elementToJaxb(request))
         .onFailure(ex -> mLog.debug(ex.getMessage()))
-        .mapFailure(Case($(instanceOf(Exception.class)),
-            ex -> ServiceException.PARSE_ERROR("Malformed request.", ex)));
+        .mapFailure(
+            Case(
+                $(instanceOf(Exception.class)),
+                ex -> ServiceException.PARSE_ERROR("Malformed request.", ex)));
   }
 
   /**
-   * Get attachment using request and context authorization info.
+   * Gets an attachment using request and context authorization info.
    *
    * @param request try of {@link CopyToFilesRequest}
    * @param context the context for current session
    * @return try of a {@link MimePart}
    */
   private Try<MimePart> getAttachmentToCopy(CopyToFilesRequest request, ZimbraSoapContext context) {
     // get mail message
-    final Try<Integer> messageIdTry = Try.of(() -> Integer.parseInt(request.getMessageId()))
-        .onFailure(ex -> mLog.debug(ex.getMessage()))
-        .mapFailure(Case($(instanceOf(Exception.class)),
-            ex -> ServiceException.PARSE_ERROR(MailConstants.A_MESSAGE_ID + " must be an integer.",
-                ex)));
-    // get mail attachment
-    return For(Try.of(() -> request), messageIdTry).yield((req, messageId) ->
-        attachmentService.getAttachment(context.getAuthtokenAccountId(), context.getAuthToken(),
-                messageId, req.getPart())
+    final Try<Integer> messageIdTry =
+        Try.of(() -> Integer.parseInt(request.getMessageId()))
             .onFailure(ex -> mLog.debug(ex.getMessage()))
-            .mapFailure(Case($(instanceOf(Exception.class)),
-                ex -> ServiceException.NOT_FOUND("File not found.", ex)))).get();
+            .mapFailure(
+                Case(
+                    $(instanceOf(Exception.class)),
+                    ex ->
+                        ServiceException.PARSE_ERROR(
+                            MailConstants.A_MESSAGE_ID + " must be an integer.", ex)));
```

Comment:
See suggestion above:
```suggestion
            .onFailure(Exception.class,
                    ex ->
                        ServiceException.PARSE_ERROR(
                            MailConstants.A_MESSAGE_ID + " must be an integer.", ex));
```